src/Form/UpdateMembershipRequestType.php

```diff
@@ -2,30 +2,55 @@
 
 namespace AppBundle\Form;
 
+use AppBundle\Membership\MembershipRequest;
 use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\BirthdayType;
+use Symfony\Component\Form\Extension\Core\Type\CheckboxType;
 use Symfony\Component\Form\Extension\Core\Type\EmailType;
 use Symfony\Component\Form\Extension\Core\Type\TextType;
 use Symfony\Component\Form\FormBuilderInterface;
 use Symfony\Component\OptionsResolver\OptionsResolver;
+use Misd\PhoneNumberBundle\Form\Type\PhoneNumberType;
 
 class UpdateMembershipRequestType extends AbstractType
 {
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
         $builder
-            ->remove($firstName = 'firstName')->add($firstName, TextType::class)
-            ->remove($lastName = 'lastName')->add($lastName, TextType::class)
-            ->remove($email = 'emailAddress')->add($email, EmailType::class)
+            ->add('gender', GenderType::class)
+            ->add('firstName', TextType::class, ['disabled' => true])
+            ->add('lastName', TextType::class, ['disabled' => true])
+            ->add('emailAddress', EmailType::class, ['disabled' => true])
+            ->add('birthdate', BirthdayType::class, [
+                'widget' => 'choice',
+                'years' => $options['years'],
+                'placeholder' => [
+                    'year' => 'AAAA',
+                    'month' => 'MM',
+                    'day' => 'JJ',
+                ],
+            ])
+            ->add('position', ActivityPositionType::class)
+            ->add('address', AddressType::class)
+            ->add('phone', PhoneNumberType::class, [
+                'required' => false,
+                'widget' => PhoneNumberType::WIDGET_COUNTRY_CHOICE,
+            ])
+            ->add('comMobile', CheckboxType::class, [
+                'required' => false,
+            ])
         ;
     }
 
-    public function getParent()
-    {
-        return MembershipRequestType::class;
-    }
-
     public function configureOptions(OptionsResolver $resolver)
     {
-        $resolver->setDefault('validation_groups', ['Default']);
+        $years = range((int) date('Y') - 15, (int) date('Y') - 120);
+
+        $resolver->setDefaults([
+            'data_class' => MembershipRequest::class,
+            'translation_domain' => false,
```

Comment:
Why?

Reply:
All labels are written directly in the view so we don't need to trigger the translator.

Reply:
I heard they want to use translation soon, I understand the value but this is not worth it anymore.